### PR TITLE
Use clang-format-11 in code-formatting checker

### PR DESCRIPTION
**Pull request by TimoWilken**
- install and use clang-format-11
- pass `--style file` to clang-format to explicitly use `.clang-format` file in project root
- minor improvements to formatting and small speed-up
- removed unused code
- added clang-format version to message on how to reproduce result

### CODE REVIEW

File: .github/workflows/code-formatting.yml [hunk 1]
```diff
@@ -15,68 +16,69 @@ jobs:
...
         # github.event.pull_request.base.sha is the latest commit on the branch
         # the PR will be merged into, NOT the commit this PR derives from! For
         # that, we need to find the latest common ancestor between the PR and
         # the branch we are merging into.
         BASE_COMMIT=$(git merge-base HEAD ${{ github.event.pull_request.base.sha }})
-        echo "Running clang-format-8 against branch ${{ github.event.pull_request.base.ref }} , with hash ${{ github.event.pull_request.base.sha }}"
         COMMIT_FILES=$(git diff --name-only $BASE_COMMIT | grep -ivE 'LinkDef|Utilities/PCG/')
-        [ "X$COMMIT_FILES" = X ] && { echo "No files to check" ; exit 0; }
-        RESULT_OUTPUT="$(git-clang-format-8 --commit $BASE_COMMIT --diff --binary `which clang-format-8` $COMMIT_FILES)"
+        if [ -n "$COMMIT_FILES" ]; then
```
Comment (ktf) [inline .github/workflows/code-formatting.yml]: Do you really need to change the rest of the logic to bump clang-format 11?

Comment (TimoWilken) [reply .github/workflows/code-formatting.yml]: OK, I've undone most of the formatting changes. Is this better?

Comment (ktf) [reply .github/workflows/code-formatting.yml]: yes. Could you move the `11` in a variable?

Comment (ktf) [reply .github/workflows/code-formatting.yml]: Alternatively, can't we simply drop the `-11`?

Comment (TimoWilken) [reply .github/workflows/code-formatting.yml]: I've put the -11 in a variable. If we drop the -11, it'll use clang-format-10 in the Ubuntu containers provided by GitHub.

Comment (ktf) [reply .github/workflows/code-formatting.yml]: Sorry for being pedantic... Actually I think:

```
update-alternatives --install /usr/bin/clang-format clang-format /usr/bin/clang-format-11 100
```

is better. This way we do not need a custom environment variable and we can simply have `clang-format`. Am I missing something?

Comment (TimoWilken) [reply .github/workflows/code-formatting.yml]: OK, sure! I've also redirected git-clang-format and removed the `--binary ...` options as they should be superfluous now.

**Pull request merged**